app/controllers/v0/gi/institution_programs_controller.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module V0
+  module GI
+    class InstitutionProgramsController < GIController
+      def autocomplete
+        render json: client.get_institution_program_autocomplete_suggestions(scrubbed_params)
```

Comment:
I'm concerned by the following:
- This is a synchronous request -`vets-api` will hold the request until the gids service responds. 
- Autocomplete endpoints tend to get called frequently

Even if the gids service is very responsive and FE devs use a sensible bounce interval for calling this endpoint, we should try to code defensively to protect `vets-api`. Have you made any considerations for performance?

Perhaps look into using [`cache_aside`](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/common/models/concerns/cache_aside.rb).

Reply:
This is definitely an existing issue that has [wreaked havoc](https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Postmortems/2019-06-10-GI-Latency.md) before, but was never addressed.

Reply:
The institution_program autocomplete will be used by future stories in vets-website.  We also create stories for updating the new institution_program and existing institution autocompletes to use `cache_aside` in a future story. That's definitely an important existing issue that'd be great to fix.